libsolidity/formal/CHC.cpp

```diff
@@ -884,9 +893,15 @@ void CHC::defineContractInitializer(ContractDefinition const& _contract)
 	addRule(smtutil::Expression::implies(initialConstraints(_contract), implicitFact), implicitFact.name);
 	setCurrentBlock(implicitConstructorPredicate);
 
+	solAssert(!m_errorDest, "");
+	m_errorDest = m_contractInitializers.at(&_contract);
```

Comment:
@mijovic another bug was when there was a failing assertion inside an internal function called at state variable declaration, which is visited in the loop below. In that case, the error condition should jump only to the end of the contract initialization (routine defined in this function), and not to the constructor summary which represents the whole deployment procedure. Since this error destination is used by function call visits, this needs to be a member of the class. In fact this made the error destination actually simpler where it's used. The important part is to assign it properly for contract initialization, inheritance order, and function visit.